rotkehlchen/chain/ethereum/modules/curve/curve_cache.py

```diff
@@ -241,172 +262,196 @@ def save_curve_data_to_cache(
         key_parts=[GeneralCacheType.CURVE_LP_TOKENS],
     )
     # Then, read the new pool mappings and save them in the global DB cache
-    globaldb_set_general_cache_values(
-        write_cursor=write_cursor,
-        key_parts=[GeneralCacheType.CURVE_LP_TOKENS],
-        values=pools_mapping.keys(),  # keys of pools_mapping are lp tokens
-    )
-    for lp_token_address, pool_info in pools_mapping.items():
-        pool_address, coins, _underlying_coins = pool_info
+    db_addressbook = DBAddressbook(db_handler=database)
+    for pool in all_pools:
+        addresbook_entries = [AddressbookEntry(
+            address=pool.pool_address,
+            name=pool.pool_name,
+            blockchain=SupportedBlockchain.ETHEREUM,
+        )]
+        if pool.gauge_address is not None:
+            addresbook_entries.append(AddressbookEntry(
+                address=pool.gauge_address,
+                name=f'Gauge for {pool.pool_name}',
+                blockchain=SupportedBlockchain.ETHEREUM,
+            ))
+        try:
+            db_addressbook.add_addressbook_entries(
+                cursor=write_cursor,
+                book_type=AddressbookType.GLOBAL,
+                entries=addresbook_entries,
+            )
```

Comment:
looking at the usage of `add_addressbook_entries()`...

If a known entry is first encountered, it raises an error at that point and does not bother to add other entries irrespective of whether they're known or not.

Reply:
Yes, thanks for pointing it out, will add a parameter to `add_addressbook_entries`